File: Models/Letra.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using Maia.Models.Abstracts;
using Maia.Models.DTO;
using Newtonsoft.Json;

namespace Maia.Models
{
    public class Letra : AnotacaoMusical
    {
        [MaxLength(5)]
        [RegularExpression("PT-BR|EN-US", ErrorMessage = "Idioma pode ser apenas 'PT-BR' ou 'EN-US'")]
        public string Idioma { get; set; }

        [Required]
        public string Texto { get; set; }
        
        public string TextoHtml { get; set; }
        
        public Letra() { }

        public Letra(LetraDTO dto)
        {
            this.UsuarioId = dto.UsuarioId;
            this.MusicaId = dto.MusicaId;
            this.Idioma = dto.Idioma;
            this.Texto = dto.Texto;
            this.TextoHtml = dto.TextoHtml;
        }
    }
}

File: Angular/src/app/shared/models/Letra.ts
import {AnotacaoMusical} from "./AnotacaoMusical";
import {Usuario} from "./Usuario";
import {Musica} from "./Musica";

export class Letra extends AnotacaoMusical {

  public idioma: string;
  public texto: string;
  public textoHtml: string;

  constructor(
    musicaId: number, usuarioId: number, idioma: string, texto: string, textoHtml: string, usuario?: Usuario, musica?: Musica, id?: number
  ) {
    super(musicaId, usuarioId, usuario, musica, id);
    this.idioma = idioma;
    this.texto = texto;
    this.textoHtml = textoHtml;
  }
}
